src/overlays/actors/ovl_Obj_Dora/z_obj_dora.c

```diff
@@ -28,104 +47,307 @@ const ActorInit Obj_Dora_InitVars = {
+    this->gongRotation.x = Math_SinS(this->gongAngle.x) * this->gongForce.y;
+    this->gongRotation.z = Math_SinS(this->gongAngle.z) * this->gongForce.z;
+}
+
+s32 ObjDora_IsHalfHourly(u16 time) {
+    f32 timeHalfHour = time;
+
+    timeHalfHour -= (CLOCK_TIME_F(0, 30) * (s32)(time / CLOCK_TIME_F(0, 30)));
+    if ((timeHalfHour < CLOCK_TIME_F(0, 1)) || (CLOCK_TIME_F(0, 29) < timeHalfHour)) {
+        return true;
+    }
+
+    return false;
+}
+
+void ObjDora_UpdateCollision(ObjDora* this, GlobalContext* globalCtx) {
+    Actor* itemDrop;
+    u16 time;
+
+    if (this->colliderTris.base.acFlags & AC_HIT) {
+        time = gSaveContext.save.time;
+        this->colliderTris.base.acFlags &= ~AC_HIT;
+        this->collisionCooldownTimer = 5;
+
+        switch (this->actor.colChkInfo.damageEffect) {
+            case DORA_DMGEFF_STRONG:
+            case DORA_DMGEFF_LIGHT:
+                if (this->actor.colChkInfo.damageEffect == DORA_DMGEFF_LIGHT) {
+                    Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_DORA_S);
+                    this->lastGongHitType = DORA_HIT_LIGHT;
+                } else {
+                    Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_DORA_L);
+                    this->lastGongHitType = DORA_HIT_STRONG;
+                }
+
+                func_800BC848(&this->actor, globalCtx, 5, 10);
+                ObjDora_SetupMoveGong(this);
+
+                if ((ObjDora_IsHalfHourly(time) == true) && (this->rupeeDropTimer == 0)) {
+                    Actor_PlaySfxAtPos(&this->actor, NA_SE_SY_TRE_BOX_APPEAR);
+                    itemDrop = Item_DropCollectible(globalCtx, &this->actor.world.pos, ITEM00_RUPEE_BLUE);
+                    itemDrop->world.rot.y = this->actor.world.rot.y;
+                    itemDrop->world.rot.y += (s32)(Rand_Centered() * 90.0f * 182.04445f);
+                    itemDrop->velocity.y = 5.0f;
+                    itemDrop->gravity = -1.0f;
+                    this->rupeeDropTimer = 40;
+                }
+                break;
+        }
+    }
 
-#endif
+    if (this->rupeeDropTimer > 0) {
+        this->rupeeDropTimer--;
+    }
 
-extern ColliderTrisElementInit D_80B61310[6];
-extern ColliderTrisInit D_80B61478;
-extern DamageTable D_80B61488;
-extern CollisionCheckInfoInit2 D_80B614A8;
+    if (this->collisionCooldownTimer > 0) {
+        this->collisionCooldownTimer--;
+    } else {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderTris.base);
+    }
+}
 
-extern UNK_TYPE D_06004160;
+void ObjDora_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjDora* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Dora/ObjDora_Init.s")
+    this->actionFunc(this, globalCtx);
+    ObjDora_UpdateCollision(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Dora/ObjDora_Destroy.s")
+void ObjDora_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    ObjDora* this = THIS;
+    f32 gongForceX;
```

Comment:
```suggestion
    static Vec3f position = { 0.0f, -61.5f, 0.0f };
    ObjDora* this = THIS;
    f32 gongForceX;
```
Generally in-function static variables are above the other declarations with no space in between.